ArcadiaTeamsBot/Dialogs/NewRequestDialog.cs

```diff
@@ -1,63 +1,276 @@
+                                .Items
+                                .Split(';')
+                                .Select(item => new AdaptiveChoice { Title = item, Value = item })
+                                .ToList()
+                        };
+                        break;
+
+                    case RequestTypeUIFieldType.Number:
+                        textBlock = new AdaptiveTextBlock("Enter a number");
+                        input = new AdaptiveNumberInput { Id = field.FieldName, Placeholder = field.FieldName };
+                        break;
+
+                    case RequestTypeUIFieldType.String:
+                        textBlock = new AdaptiveTextBlock("Enter a string");
+                        input = new AdaptiveTextInput { Id = field.FieldName, Placeholder = field.FieldName };
+                        break;
+
+                    default:
+                        textBlock = new AdaptiveTextBlock("Unknown field type. Enter a string");
+                        input = new AdaptiveTextInput { Id = field.FieldName, Placeholder = field.FieldName };
+                        break;
+                }
+
+                cardBody.Add(textBlock);
+                cardBody.Add(input);
             }
-            return await stepContext.ContinueDialogAsync(cancellationToken: cancellationToken);
+
+            var Actions = new List<AdaptiveAction>
+            {
+                new AdaptiveSubmitAction
+                {
+                    Title = Submit,
+                    Data = new { Button = Submit }
+                },
+                new AdaptiveSubmitAction
+                {
+                    Title = Cancel,
+                    Data = new { Button = Cancel }
+                }
+            };
+
+            var promptOptions = new PromptOptions
+            {
+                Prompt = (Activity)MessageFactory.Attachment(InputFormCard(cardBody, Actions)),
+                RetryPrompt = MessageFactory.Text("Not all fields are filled. Repeat")
+            };
+
+            await stepContext.Context.SendActivityAsync(promptOptions.Prompt, cancellationToken);
+            promptOptions.Prompt = new Activity(ActivityTypes.Message);
+            return await stepContext.PromptAsync("askForInput", promptOptions, cancellationToken);
         }
 
-        public static HeroCard GetInfoCard()
+        private async Task<DialogTurnResult> ChoiceStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
         {
-            var infoCard = new HeroCard
+            var formData = (JObject)stepContext.Context.Activity.Value;
+            if (formData["Button"].ToString() == Cancel)
             {
-                Title = "In Development",
-                Buttons = new List<CardAction>
+                return await stepContext.ReplaceDialogAsync(nameof(MainDialog), null, cancellationToken);
```

Comment:
Probably, it is better to send user to previous dialog instead of main dialog in such case.

Reply:
Fixed